src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFCD.cs

```diff
@@ -472,15 +471,16 @@ internal static string AnchorDescription(int anchors)
 #endif
     }
 
-    internal sealed class RegexFC
+    internal struct RegexFC : IDisposable
```

Comment:
IDisposable may not be a good fit for a struct especially when forgetting to dispose is not a resourcel eak.

https://github.com/dotnet/corefx/pull/27250#discussion_r169745944

Reply:
You are not using it either you are calling Dispose directly.

Reply:
it's not a ref type but I agree, we don't need it here. I had a different implementation for ValueListBuilder which detects if the type implements IDisposable and then calls it but there was a perf regression with the type is IDisposable check (reflection).